pkg/manifests/manifests.go

```diff
@@ -806,6 +825,44 @@ func (f *Factory) KubeStateMetricsPrometheusRule() (*monv1.PrometheusRule, error
 	return f.NewPrometheusRule(f.assets.MustNewAssetReader(KubeStateMetricsPrometheusRule))
 }
 
+func (f *Factory) KubeStateMetricsCRSConfigMap() (*v1.ConfigMap, error) {
+	configmap, err := f.NewConfigMap(f.assets.MustNewAssetReader(KubeStateMetricsCRSConfig))
+	if err != nil {
+		return nil, err
+	}
+
+	// Check CRS prerequisites before enabling CRS metrics in CMO.
+	// Currently, the prerequisites are:
+	// * The presence of `VerticalPodAutoscaler` CRD in the cluster: `KubeStateMetricsListErrors` will be fired if absent.
+	_, err = f.client.ApiExtensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), verticalPodAutoscalersGR, metav1.GetOptions{})
```

Comment:
It breaks the (implicit) separation of concerns that Factory doesn't interact with the Kubernetes API.